src/components/Sidebar/BoardLink/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { FC, memo, useCallback } from 'react'
+import { Link } from 'react-router-dom'
+
+import styled from 'styled-components'
+
+import { ServicePath } from '../../../enums/ServicePath'
+import { useAppDispatch } from '../../../hooks/useAppDispatch'
+import { setCurrentBoardId } from '../../../store/reducers/boardsReducer'
+import { BasicMenu } from '../../../common/Menu'
+import { LabelMessage } from '../../../enums/Message'
+
+type PropsType = {
+  active: boolean
+  id: number
+  title: string
+  onDeleteButtonClick: (id: number) => void
+}
+
+export const BoardLink: FC<PropsType> = memo(({ active, id, title, onDeleteButtonClick }) => {
+  const dispatch = useAppDispatch()
+
+  const deleteClickHandler = useCallback(() => {
+    onDeleteButtonClick(id)
+  }, [id, onDeleteButtonClick])
+
+  const onClickHandler = () => dispatch(setCurrentBoardId({ id: +id }))
```

Comment:
not sure that we have to transform id to number, because in prop type we already expect that id is number